code-challenges.test.js

```diff
@@ -10,41 +19,181 @@
 
-const dataArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
-// Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+describe("combineArrays", () => {
+  const dataArray1 = ["array", "object", "number", "string", "Boolean"];
+  const dataArray2 = ["string", "null", "Boolean", "string", "undefined"];
+
+  it("takes in two arrays as arguments and returns one array with no duplicate values.", () => {
+    expect(combineArrays(dataArray1, dataArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"]);
+  });
+});
 
 
+
+// Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+
 // b) Create the function that makes the test pass.
+// create a function called combineArrays with a parameter of arrays
+//Set up a new array to hold the desired values.
+// combine the arrays into a single array using a spread operator, and sort them. This will place any duplicates next to each other in the array, making it possible to check each one to its neighbor.
+// for each item in the array, check if its value matches that of the next item. If not (meaning it is not a duplicate) add it to the new array.
+// Return the new array
+
+// Another option I considered was to splice off the duplicate from the original array and return it, but it seemed to mess with the logic of iterating through the array when I was splicing items out.
+
+
+// const combineArrays2 = (a, b) => {
+//     newArrA = [...a, ...b].sort();
+//     newArrB = []
+//     for (let i = 0; i < newArrA.length; i++) {
+//       if (newArrA[i] !== newArrA[i+1])
+//       newArrB = [...newArrB, newArrA[i]]  
+//     }
+//     return newArrB
+//   }
+//   expect(received).toEqual(expected) // deep equality
+
+//   - Expected  - 3
+//   + Received  + 3
+
+//     Array [
+//   +   "Boolean",
+//       "array",
+//   -   "object",
+//   +   "null",
+//       "number",
+//   +   "object",
+//       "string",
+//   -   "Boolean",
+//   -   "null",
+//       "undefined",
+//     ]
+
+//------------This solution works for the prompt as written (does not specify order), but returns the string in a different order than expected, so it does not pass the test. I chose not to use the array contains function from the first challenge, since it defeats the purpose of th prompt, to remove extras.
+
+
+// I came up with a new solution that will keep the original order. 
+// combine the arrays as before. Then create a for loop.
+// for each index in the array, check if it is also the last index of that item. 
+// as long as that is NOT true, keep splicing off the item at the last index until the first and last index are equal. Then return the array.
+
+const combineArrays = (a, b) => {
+    let one = [...a, ...b];
+    for (let i = 0; i < one.length; i++) {
+      while (i !== one.lastIndexOf(one[i])) {
```

Comment:
I like that you are practicing while loops, can you think of a way to approach this problem without using for loops though? (Hint: research new Set)